asf_core_data/pipeline/mcs/process/process_historical_mcs_installers.py

```diff
@@ -0,0 +1,926 @@
+    Computes full_address by putting together the different lines in address
+    (address_1, address_2, town, county and postcode) for original records.
+
+    Args:
+        address_1: first line of address
+        address_2: second line of address
+        town: town
+        county: county
+        postcode: postcode
+        original_record: True if record originally in installers table
+    Returns:
+        full adress
+    """
+    if original_record:
+        temp = address_1
+        if not pd.isnull(address_2):
+            temp = temp + address_2 + ", "
+        if not pd.isnull(town):
+            temp = temp + town + ", "
+        if not pd.isnull(county):
+            temp = temp + county + ", "
+        if not pd.isnull(postcode):
+            temp = temp + postcode + ", "
+        return temp[:-2]
+    # if not original record, we have the full address from Companies House
+    return full_address
+
+
+def create_certified_flags(data: pd.DataFrame):
+    """
+    Updates historical installers data by creating flags representing whether
+    installer is certified or not for a certain technology.
+
+    If a start date exists, then if means installer is certified.
+
+    Args:
+        data: historical installers data
+    """
+
+    # start date variables for all tech types e.g. air_source_hp_start_date
+    start_date_cols = [col for col in data.columns if "start_date" in col]
+
+    for col in start_date_cols:
+        # example of flag_var: air_source_hp_certified
+        flag_var = col.replace("start_date", "certified")
+        data[flag_var] = ~pd.isnull(data[col])
+
+
+# copied from process_mcs_utils.py and changed slightly
+def clean_company_name(company_name: str) -> str:
+    """
+    Cleans installation company name by:
+        - changing company name to lower case;
+        - removing punctuation;
+        - removing company-related stopwords like "ltd", "limited", etc.;
+
+    Args:
+        company_name: Name of company to clean.
+    Returns:
+        company_name: Cleaned company name.
+    """
+    company_stopwords = ["ltd", "limited", "old", "account"]
+
+    company_name = company_name.lower().translate(str.maketrans("", "", "()-,/.&"))
+    company_name = [
+        comp for comp in company_name.split() if comp not in company_stopwords
+    ]
+
+    return " ".join(company_name)
+
+
+def from_list_to_dictionary(list_values):
+    """
+    Transforms a list into a dictionary where each key is matched to the first
+    value in the list. Creates a mapping between each item in the list and the first one.
+    Arg:
+        list_values: a list of values e.g. ["Company A", "Company B", "Company A trading as Company B"]
+    Returns:
+        Dictionary mapping all values to the first one
+        e.g. {"Company B":"Company A", "Company A trading as Company B": "Company A"}
+    """
+    dictionary = dict()
+    for value in list_values[1:]:
+        dictionary[value] = list_values[0]
+    return dictionary
```

Comment:
Why would we need this...? I'll probably find out later but it seems odd.  

Either way, you can do this much simpler (see also dummy example). Hardly worth having it's own function. 

![Screenshot 2023-02-25 at 17 04 23](https://user-images.githubusercontent.com/42718928/221369980-8db8ead6-1d46-411c-8100-e1561f28511c.png)

```suggestion
def from_list_to_dictionary(list_values):
    """
    Transforms a list into a dictionary where each key is matched to the first
    value in the list. Creates a mapping between each item in the list and the first one.
    Arg:
        list_values: a list of values e.g. ["Company A", "Company B", "Company A trading as Company B"]
    Returns:
        Dictionary mapping all values to the first one
        e.g. {"Company B":"Company A", "Company A trading as Company B": "Company A"}
    """
    return {value : list_values[0] for value in list_values[1:]}
```